src/rpcblockchain.cpp

```diff
@@ -1518,14 +1518,46 @@ Value execute(const Array& params, bool fHelp)
         entry.push_back(Pair("My Neural Hash",myNeuralHash.c_str()));
         results.push_back(entry);
     }
-    else if (sItem == "superblockage")
-    {
-        int64_t superblock_time = ReadCache("superblock", "magnitudes").timestamp;
-        int64_t superblock_age = GetAdjustedTime() - superblock_time;
-        entry.push_back(Pair("Superblock Age", superblock_age));
-        entry.push_back(Pair("Superblock Timestamp", TimestampToHRDate(superblock_time)));
-        entry.push_back(Pair("Superblock Block Number", ReadCache("superblock", "block_number").value));
-        entry.push_back(Pair("Pending Superblock Height", ReadCache("neuralsecurity","pending").value));
+    else if (sItem == "superblockinfo")
+    {
+        std::string sSbHeight = ReadCache("superblock", "block_number").value;
+        int nSbHeight;
+
+        try
+        {
+            nSbHeight = std::stoi(sSbHeight);
+        }
+
+        catch (std::exception &ex)
+        {
+            entry.push_back(Pair("ERROR", "std::exception occured ->" + std::string(ex.what())));
+            results.push_back(entry);
+
+            return results;
+        }
+
+        CBlockIndex* pblockindex = RPCBlockFinder.FindByHeight(nSbHeight);
+
+        if (pblockindex == NULL)
+            entry.push_back(Pair("ERROR", "Could not find superblock in blockindex"));
+
+        else
+        {
+            CBlock block;
+
+            block.ReadFromDisk(pblockindex);
+
+            MiningCPID sbi = DeserializeBoincBlock(block.vtx[0].hashBoinc, block.nVersion);
+
+            entry.push_back(Pair("age", GetAdjustedTime() - pblockindex->GetBlockTime()));
+            entry.push_back(Pair("time", pblockindex->GetBlockTime()));
+            entry.push_back(Pair("height", nSbHeight));
+            entry.push_back(Pair("hash", pblockindex->GetBlockHash().ToString()));
+            entry.push_back(Pair("neural_hash", sbi.NeuralHash));
+            entry.push_back(Pair("CPID", sbi.cpid));
```

Comment:
Was this different from `pblockindex->GetCPID()`?

Reply:
yea `pblockindex->GetCPID()` gave all zeros for investor and `sbi.cpid` gave correct cpid. strange and odd i agree.

Reply:
When things are supposed to be equal, I always add and an `assert`ion even if the code does not depend on it.